telosnetwork/telos-wallet
# Integration of new layout with new pinia store

# Fixes #152, #170, #171
## also fixes #205 and #203 (inherited from [PR219](https://github.com/telosnetwork/telos-wallet/pull/219))

## Description
This PR integrates the latest visual components refactoring work (from [PR218](https://github.com/telosnetwork/telos-wallet/pull/218) and [PR193](https://github.com/telosnetwork/telos-wallet/pull/193) @EJSG's work) on the EVM wallet's user interface, along with the new implementation of the Pinia store-based library (from [PR219](https://github.com/telosnetwork/telos-wallet/pull/219) and [PR211](https://github.com/telosnetwork/telos-wallet/pull/211)).

The site's navigation now includes user authentication, along with some basic functionalities such as displaying the logged-in account's address, copying it to the clipboard with a click, and displaying a menu with options to redirect to Telosscan, log out, or switch to dark mode.

As for the native part, there shouldn't be any changes beyond the new way of authenticating from the new HomePage.

**Note:** since [PR219](https://github.com/telosnetwork/telos-wallet/pull/219) is still not merged but is included in this PR I will close PR219 and include the issues index in this PR

## Test scenarios
- go to [this link](https://deploy-preview-220--wallet-staging.netlify.app/)
- press **Connect your wallet button**
  - The MetaMask wallet should popup
  - if you don't have the network configured, it will prompt you to add it
  - if you are not connected to that chain it will ask you to change the current chain
  - if you are not authenticated you will be asked to authenticate
  - If you finally log in, you will be redirected to `/evm/wallet?tab=balance` (**new App Skeleton**)
- once logged in, you should see your real address displayed on the top-right
- if you press the button next to the address
  - it should copy the complete address to your clipboard
  - A notification should appear saying that
- If you press the account dropdown button, a menu should appear with three options:
  - teloscan: which redirects you to a teloscan page of your logged account address
  - logout: which actually logs you out and redirects you to the home page
  - dark mode: which switched between dark and light modes 
- autologin: this works automatically and should redirect you to the correct page
  - if you are logged in, go to [homepage](https://deploy-preview-220--wallet-staging.netlify.app/)
     - it should redirect you to  `/evm/wallet?tab=balance` 
  - if you are logged out, go to any `/evm/...` path link [this link](https://deploy-preview-220--wallet-staging.netlify.app/evm/wallet?tab=balance) or `[this link](https://deploy-preview-220--wallet-staging.netlify.app/evm/wrap?tab=unwrap)`
    - then you should be redirected to the Home page

## Checklist:
